src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,110 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.connectionutil.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final int FIRST_INDEX = 1;
+    private static final int SECOND_INDEX = 2;
+    private static final int THIRD_INDEX = 3;
+    private static final int ZERO_ROW = 0;
+    private static final String INSERT_SQL = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String SELECT_BY_ID_SQL = "SELECT * FROM books WHERE id = ?";
+    private static final String SELECT_ALL_SQL = "SELECT * FROM books";
+    private static final String UPDATE_SQL = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_SQL = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(
+                        INSERT_SQL, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(FIRST_INDEX, book.getTitle());
+            statement.setBigDecimal(SECOND_INDEX, book.getPrice());
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(FIRST_INDEX, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add a new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(UPDATE_SQL)) {
+            statement.setString(FIRST_INDEX, book.getTitle());
+            statement.setBigDecimal(SECOND_INDEX, book.getPrice());
+            statement.setLong(THIRD_INDEX, book.getId());
+            statement.executeUpdate();
+            return findById(book.getId()).orElse(book);
```

Comment:
Generally, it's a good practice to fetch a most actual data from DB after an update, but this part `.orElse(book)` I don't understand clearly. If book with the same ID after update is not found, I think it should be exceptional case, but you're returning the book from request.